pool/src/pool/submit.rs

```diff
@@ -63,6 +61,84 @@ pub fn execute_submit(
     new_from_state.positions
 }
 
+// Note: if this looks like a good approach we should probably refactor the code
+// a bit to prevent duplicating execute_submit.
+pub fn execute_submit_with_flash_loan(
+    e: &Env,
+    from: &Address,
+    flash_loan: FlashLoan,
+    requests: Vec<Request>,
+    use_allowance: bool,
+) -> Positions {
+    if from == &e.current_contract_address()
+    {
+        panic_with_error!(e, &PoolError::BadRequest);
+    }
+    let mut pool = Pool::load(e);
+
+    // note: check_health is omitted since we always will want to check the health
+    // if a flash loan is involved.
+    let (actions, mut new_from_state, _) =
+        build_actions_from_request(e, &mut pool, from, requests);
+
+    // similarly to usual borrows, we want the flash loan to be debited before
+    // checking the health factor. TODO: We should decide whether the execution ordering
+    // matters here (I'm pretty sure it doesn't but haven't looked much into it).
```

Comment:
This would only be an issue if the function ever was re-entrant.

maybe make a comment that re-entrancy is unsafe here since the user state is not written before the flash loan contract was invoked, to warn any future developer.